index.js

```diff
@@ -11,50 +11,101 @@ var couch2pg = require('./libs/couch2pg/importer')(
       db, couchdb,
       env.couch2pgDocLimit,
       env.couch2pgChangesLimit),
-    xmlforms = require('./libs/xmlforms/extractor')(db);
-
-var batchLoop = function() {
-  var completeBatch = function(lastSeq) {
-    return couch2pg.storeSeq(lastSeq)
-      .then(function() {
-        log.info('Batch completed up to ' + lastSeq);
-      });
-  };
-
-  log.info('Beginning couch2pg and xmlforms batch');
-
-  return couch2pg.importBatch()
-  .then(function(changes) {
-    var allDocs = changes.deleted.concat(changes.edited);
-    log.debug('There are ' + allDocs.length + ' changes');
-
-    if (allDocs.length > 0) {
-      return xmlforms.extract(allDocs)
-      .then(function() {
-        return completeBatch(changes.lastSeq);
-      })
-      .then(batchLoop);
+    xmlforms = require('./libs/xmlforms/updater')(db);
+
+var fallback = 0;
+var sleepMs = function(errored) {
+  if (errored) {
+    var fallbackMs = fallback * 1000 * 60;
+    if (fallbackMs < env.sleepMs) {
+      fallback++;
+      return fallbackMs;
+    } else {
+      return env.sleepMs;
+    }
+  } else {
+    fallback = 0;
+    return env.sleepMs;
+  }
+};
+
+var migrateCouch2pg = function() {
+  return couch2pgMigrator(env.postgresqlUrl)();
+};
+var migrateXmlforms = function() {
+  return xmlformsMigrator(env.postgresqlUrl)();
+};
+
+var delayLoop = function(loopFn, errored) {
+  return new Promise(function(resolve) {
+    var ms = sleepMs(errored);
+    log.info('Run '+(errored ? 'errored' : 'complete') + '. Next run at ' + new Date(new Date().getTime() + ms));
+    if (ms === 0) {
```

Comment:
Hmm, your code wouldn't do the same thing, so this is definitely too confusing!

Basically this is what's happening:
 - Calculate how long to wait (`sleepMs()`):
   - If there was no error it's whatever the sleep time is
   - If there was an error sleep for a amount of time based on a falloff period, so that if there is an error it retries immediately and then waits longer and longer with subsequent errors

Because both erroring loops and successful loops can wait, we need to calculate the ms in both instances. 

Let me know if that makes sense, and how we change the code to make that clearer.